lib/puppet/module/task.rb

```diff
@@ -61,6 +62,29 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split(File::SEPARATOR)
+          pup_module = Puppet::Module.find(splits[0])
+          if pup_module.nil?
+            warning _("Could not find module %{module}" % { module: splits[0] })
+            return []
+          end
+          mount = splits[1]
+          unless MOUNTS.include? mount
+            msg = _("Files must be saved in module directories that Puppet makes available via mount points: files, lib, tasks")
+            raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+          end
+          path = File.join(pup_module.path, mount, splits[2..-1])
+
+          { "name" => name, "path" =>  path }
```

Comment:
`name` here is the name of the module, isn't it? But for implementation, it's the basename of the file. There's definitely an assumption that the `name` is unique, but that wouldn't necessarily be the case if we just used the basename. But at the same time, the reason we have `name` for implementations is so that we can look up the path. These "extra" files won't be looked up by name that way, just downloaded. So maybe they should be treated separately?

Reply:
I'm mostly not sure how 'name' is used in other tools that use the task information service. Should we make name the name of the task or module that has 'files' defined?  Should we just make it empty or not have it?

Reply:
I think `name` is processed by puppetserver. It's not part of metadata, and it doesn't appear to directly make it into the `files` list Orchestrator sends to pxp-agent.